react/react1/week2/React - ONE/vite-project/src/App.jsx

```diff
@@ -19,32 +32,62 @@ function App() {
   const addTask = () => {
     const task = {
       id: todoList.length === 0 ? 1 : todoList[todoList.length - 1].id + 1,
-      taskName: newTask
-    }
+      taskName: newTask,
+      completed: false,
+    };
     setTodoList([...todoList, task]);
   };
 
   const deleteTask = (id) => {
-  
     setTodoList(todoList.filter((task) => task.id !== id));
   };
 
+  const completeTask = (id) => {
+    setTodoList(
+      todoList.map((task) => {
+        if (task.id === id) {
+          return { ...task, completed: true };
+        } else {
+          return task;
+        }
+      })
+    );
+  };
+
   return (
     <div className="App">
-    <div className='addTask'>
-    <input onChange={handleChange}/>
-    <button onClick={addTask}> Add task</button>
-    <div className='List'>
-      {todoList.map((task) => {
-        return (
-          <div>
-          <h1>{task.taskName}</h1>
-          <button onClick={() => deleteTask(task.id)}>Delete Task</button>
-          </div>
-          );
-      })}
-      </div>
+    <h1>First and upmost todo-List!</h1>
+    <img
+        src={productivityImageURL}
+        alt="Productivity"
+        className="rounded-image"
+      />
+      <div className="addTask">
+        <input className="input-1" onChange={handleChange} />
+        <button className="add-button" onClick={addTask}>
+          Add task
+        </button>
+        <div className="List">
+          {todoList.length === 0 ? (
+            <h1>You have no tasks for today. <br></br>Create one or enjoy your free time! :)</h1>
+          ) : (
+            todoList.map((task) => (
+              <Card
+                taskName={task.taskName}
```

Comment:
when you render a list in the component. you need to add a key prop to each item in the list . and the key should be a unique. so you can add fx key={task.id} here

Reply:
Thank you! Merging it now.